Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Beckn/Update.hs

```diff
@@ -169,10 +170,14 @@ handler (UEditLocationReq EditLocationReq {..}) = do
         (pickedWaypoints, currentPoint) <-
           if ride.status == DRide.INPROGRESS
             then do
-              locationPts <- LTS.driverLocation rideId merchantOperatingCity.merchantId ride.driverId
-              let (currentPoint :: Maps.LatLong) = last locationPts.loc
-              pts <- getInterpolatedPointsImplementation ride.driverId
-              return (srcPt :| (pickWaypoints pts ++ [currentPoint, dropLatLong]), Just currentPoint)
+              currentLocationPointsBatch <- LTS.driverLocation rideId merchantOperatingCity.merchantId ride.driverId
+              previousLocationPointBatches <- getInterpolatedPointsImplementation ride.driverId
+              let (currentPoint :: Maps.LatLong) =
+                    lastMaybe currentLocationPointsBatch.loc
+                      <|> lastMaybe previousLocationPointBatches
+                      <|> ((ride.fromLocation <&> (.lat)) <*> (ride.fromLocation <&> (.lon)) <&> Maps.LatLong)
+                      & fromMaybeM (InternalError "Not able to find current location")
+              return (srcPt :| (pickWaypoints previousLocationPointBatches ++ [currentPoint, dropLatLong]), Just currentPoint)
```

Comment:
`return (srcPt :| (pickWaypoints (previousLocationPointBatches <> currentLocationPointsBatch) ++ [currentPoint, dropLatLong]), Just currentPoint)`

@RitikaHanish Should'nt it be like this. We need to consider points from current batch as well right while calling pickWayPoints ?